### PR TITLE
Convert edit user page to react

### PR DESCRIPTION
**Pull request by grafixeyehero**
Depend on  convert NewUserPage to react #3026

### CODE REVIEW

**Comment by grafixeyehero** [comment]
> 
> 
> It looks like this will need rebased since your other PR has been merged.

This is my first rebase

File: src/components/dashboard/users/LnkEditUserPreferences.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    __html: `<a
+        is="emby-linkbutton"
+        class="${className}"
+        href='#'
+        >
+        ${title}
+    </a>`
+});
+
+const LnkEditUserPreferences: FunctionComponent<IProps> = ({ className, title }: IProps) => {
```
Comment (thornbill) [inline src/components/dashboard/users/LnkEditUserPreferences.tsx:19]: It looks like this should be "Link". The file name has the same typo.

File: src/components/dashboard/users/SelectElement.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        >
+        ${option}
+    </select>`
+});
+
+type IProps = {
+    className?: string;
+    label?: string;
+    currentProviderId: string;
+    providers: any
```
Comment (thornbill) [inline src/components/dashboard/users/SelectElement.tsx:18]: The type should probably be `any[]`

**Review by thornbill — CHANGES_REQUESTED** [review]
I have not finished reviewing this, but here are a couple items I noticed after a first pass.

File: src/components/dashboard/users/LnkEditUserPreferences.tsx (hunk 1 above)
Comment (grafixeyehero) [reply src/components/dashboard/users/LnkEditUserPreferences.tsx]: https://github.com/jellyfin/jellyfin-web/pull/3070/commits/1a66afa6104ce9c68bac300626a35f5d1426a15f

File: src/components/dashboard/users/SelectElement.tsx (hunk 2 above)
Comment (grafixeyehero) [reply src/components/dashboard/users/SelectElement.tsx]: https://github.com/jellyfin/jellyfin-web/pull/3070/commits/1a66afa6104ce9c68bac300626a35f5d1426a15f

File: src/components/dashboard/users/TabLinkElement.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// eslint-disable-next-line eslint-comments/disable-enable-pair
```
Comment (thornbill) [inline src/components/dashboard/users/TabLinkElement.tsx:1]: This should be removed and the rule enabled at the end of the file.

**Review by thornbill — CHANGES_REQUESTED** [review]
In addition to the one comment here, I noticed a few issues when testing.
 
- [ ] Tabs are not styled correctly when on the "Profile" tab
- [ ] Spacing seems a bit tighter than it was previously particularly in the checkbox lists
- [x] The section for allowing media deletion has some issues
  - All libraries checkbox should be included in list of libraries
  - `DeleteFoldersAccess` and `DeleteFoldersAccessHelp` sections should be removed

**Review by thornbill — APPROVED** [review]

**Pull request merged**